docs/source/configuration/authorization.mdx

```diff
@@ -98,199 +165,278 @@ So the `me` top level operation would not even be executed. When returning the r
+If every field on a particular subgraph query requires scopes that aren't present, this can eliminate entire subgraph requests.
 
-Depending on which scope set is presented by the request, different parts of the query may be available.
+#### Prerequisites
 
-Assuming we have this schema:
+To use the `@requiresScopes` directive in a subgraph you can [import it from the `@link` directive](/federation/federated-types/federated-directives/#importing-directives) like so:
 
 ```graphql
+extend schema
+  @link(
+    url: "https://specs.apollo.dev/federation/v2.5",
+    import: [..., "@requiresScopes"])
+```
+
+#### Example `@requiresScopes` use case
+
+Imagine your social media platform lets users view other users' information only if they have the required permissions.
+
+Your schema may look something like this:
+
+```graphql title=""
 type Query {
-    me: User @requiresScopes(scopes: ["profile:read"])
-    products: [Products]
+  me: User @authenticated
+  user(id: ID!): User @requiresScopes(scopes: ["read:others"])
+  users: [User!]! @requiresScopes(scopes: ["read:others"])
 }
 
 type User {
-    id: ID
-    name: String
-    email: @requiresScopes(["user:read"])
+  id: ID!
+  name: String
+  email: String @requiresScopes(scopes: ["read:email"])
+  profileImage: String
+  posts: [Post!]!
 }
 
-type Product {
-    id: ID
-    name: String
-    amount: Int @requiresScopes(["employee", "inventory"])
-    orders: [User] @requiresScopes(["employee"])
+type Post {
+  id: ID!
+  author: User!
+  title: String!
+  content: String!
+  views: Int @authenticated
 }
 ```
 
-And this query:
+The router executes the following query differently, depending on the request's attached scopes.
 
-```graphql
-{
-    me {
-        name
-        email
-    }
+If the request includes only the `read:others` scope, then the router would execute the following filtered query:
 
-    products {
-        name
-        amount
-        orders {
-            name
-        }
-    }
+<CodeColumns>
+
+```graphql title="Raw query to router"
+query {
+  users {
+    name
+    profileImage
+    email
+  }
 }
 ```
 
-If the request presented the scopes `profile:read user:read`, then it would be interpreted as:
+```graphql title="Scopes: 'read:others'"
+query {
+  users {
+    name
+    profileImage
+  }
+}
+```
 
+</CodeColumns>
 
-```graphql
-{
-    me {
-        name
-        email
-    }
+The response would include errors at the `/users/@/email` path since that field requires the `read:emails` scope.
 
-    products {
-        name
-    }
-}
-```
+If the request includes the `read:others read:emails` scope set, then the router could successfully execute the entire query.
 
-And generate an error at path `/products/@/amount`.
+<ExpansionPanel title="Want to see more examples?">
 
-While if it presented the scopes `employee inventory`, it would filter the query as:
+Take the [authorization demo](https://apollo-router-auth-demo.fly.dev/graphql) for a whirl!
```

Comment:
@lleadbet , any objections to linking the demo in the docs? I feel like it's a really great learning tool.

Reply:
Moving to Slack- will nudge you in the channel since there's some non-public info there.